### PR TITLE
adding buy button example

### PR DESCRIPTION
**Pull request by jaswinder6991**
Creating a simple buy button example. Expecting a lot of feedback.

### CODE REVIEW

File: components/Products.tsx [hunk 1]
```diff
@@ -42,7 +47,24 @@ const FETCH_STORE = gql`
...
+    // wallet?.makeOffer(tokenID,tokenPrice,{ marketAddress: process.env.marketAddress})
+    wallet?.makeOffer(tokenID,tokenPrice)
+  }
+  return buy;
+}
+
+const NFT = ({ media, title, tokens }: { media: string; title: string; tokens: Token[] }) => {
+  //const NFT = ({ media, title }: { media: string; title: string}) => {
+  const { wallet, isConnected, details } = useWallet();
+
```
Comment (microchipgnu) [inline components/Products.tsx:65]: I would do something like this:


```ts
const token = tokens[0]
const list = token?.list.price

// Maybe check if list is not null

const buy = useBuy(tokens[0]['id'],tokens[0].list.price) ;

```

File: components/Products.tsx [hunk 2]
```diff
@@ -42,7 +47,24 @@ const FETCH_STORE = gql`
   }
 `
 
-const NFT = ({ media, title }: { media: string; title: string }) => {
+const useBuy = (tokenID: string, tokenPrice: string) => {
+  const { wallet } = useWallet();
+  const tokenPriceNumber = Number(tokenPrice) ;
+  tokenPrice = (tokenPriceNumber).toLocaleString('fullwide', {useGrouping:false})
```
Comment (microchipgnu) [inline components/Products.tsx:53]: Can you explain this line here?

**Review by microchipgnu — COMMENTED** [review]
First of all, thank you so much for your PR. Looks very good.

Besides my requested changes in the comments, I also suggest removing `package-lock.json` and sticking to using yarn.

Let's change this to merge it in 🚢 🤘

File: components/Products.tsx (hunk 2 above)
Comment (jaswinder6991) [reply components/Products.tsx:53]: Line 52: Converts string into a number. But it is still represented as 1e+24.
Line 53: Converts 1e+24 to 1000000(24 times) and then convert it to a string again for make_offer to consume.

File: components/Products.tsx (hunk 1 above)
Comment (jaswinder6991) [reply components/Products.tsx:65]: So this is one of the things I am not very sure about, meaning I have been told on Stackoverflow that you should not use hooks conditionally. Hence I don't wanna put useBuy() inside an if. However this won't break cause if you look at Query on line: 22. I have put the condition "where: { storeId: { _eq: $storeId }, burnedAt: { _is_null: true }, list: {removedAt: {_is_null: true}}}"
So NFTs which are not listed won't show up only. I am interested in doing this in a better way as well.

**Comment by jaswinder6991** [comment]
> First of all, thank you so much for your PR. Looks very good.
> 
> I really like the idea of creating a hook for the buy action. This might be the start of it.
> 
> Besides my requested changes in the comments, I also suggest removing `package-lock.json` and sticking to using yarn.
> 
> Let's change this to merge it in 🚢 🤘

**Comment by jaswinder6991** [comment]
I am waiting for the conditional hook thing. I have removed package-lock.json
I can push new changes.

File: components/Products.tsx (hunk 1 above)
Comment (microchipgnu) [reply components/Products.tsx:65]: I'm sorry, I may have not explained myself correctly. You're correct, ideally you would use a hook conditionally. 

I meant that the conditional logic should be inside the hook. Thinking again about this and I propose changing the hook to take just a `thingId` argument for now and resolve all the information inside of it. 

I think this query might help with that:

```ts
  query GetMarketListedTokens($thingId: String) {
    thing(where: { id: { _eq: $thingId } }, limit: 1) {
      metadata {
        title
        description
      }
      tokens(
        where: {
          _and: [
            { list: { autotransfer: { _eq: true } } }
            { list: { removedAt: { _is_null: true } } }
            { list: { createdAt: { _is_null: false } } }
          ]
        }
        order_by: { list: { price: asc } }
      ) {
        id
        list {
          id
          price
        }
      }
    }
  }
```

With this query you're essentially fetching all tokens that are listed in the market of one thing.

Rule of thumb is: Store -> Thing -> Tokens

File: components/Products.tsx (hunk 2 above)
Comment (microchipgnu) [reply components/Products.tsx:53]: That makes sense, but it's not conventional. Have you taken a look at https://near.github.io/near-api-js/modules/utils_format.html ?

**Comment by microchipgnu** [comment]
@jaswinder6991 feel free to push the new changes.

File: components/Products.tsx (hunk 1 above)
Comment (jaswinder6991) [reply components/Products.tsx:65]: const buy = useBuy(thingId: string)
This makes a lot of sense, condition inside the hook is better than rendering it conditionally. I think if tokens array has length=0, that means there is no token of that thing listed. Do I need to use the above query?  But I think I can very easily incorporate the parameters you are using the above query in my own. Thanks. This make a lot of sense.

File: components/Products.tsx (hunk 2 above)
Comment (jaswinder6991) [reply components/Products.tsx:53]: Right, I didn't think to explore any official NEAR way to do it. In the first look it seems I can use "parseNearAmount" function. I will try this and run some tests.

**Pull request closed without merging**